src/main/java/com/google/cloud/genomics/dataflow/readers/bam/BAMShard.java

```diff
@@ -106,7 +106,6 @@ public long approximateSizeInBytes() {
 
   @Override
   public String toString() {
-    return file + ": " + contig.toString() + ", locus size = " + sizeInLoci() + 
-        ", span size = " + approximateSizeInBytes();
```

Comment:
approximateSizeInBytes updates a cached value (as does span.approximateSizeInBytes), so calling toString was causing the object to be mutated. This seemed like a reasonable solution to me.